cfgmgr/vxlanmgr.cpp

```diff
@@ -600,22 +1041,170 @@ void VxlanMgr::clearAllVxlanDevices()
+                        k.c_str(), field.c_str(), value.c_str());
+                if (field == "src_ip")
+                {
+                    src_ip = value;
+                    SWSS_LOG_NOTICE("RESTORE Vxlan Tunnel Table src_ip: %s", src_ip.c_str());
+                }
+            }
         }
-        else if (device_name.find(VXLAN_IF_NAME_PREFIX) == 0)
+        else
         {
-            info.m_vxlanIf = device_name;
-            cmdDeleteVxlanIf(info, res);
+            SWSS_LOG_NOTICE("RESTORE VxLAN Tunnel Table Key(%s)", k.c_str());
         }
     }
+
+    Table vxlanAppTunnelMapTable = Table(this->m_app_db, APP_VXLAN_TUNNEL_MAP_TABLE_NAME);
+    std::vector<std::string>::iterator it;
+    for (it = m_appVxlanTunnelMapKeysRecon.begin();
+         it != m_appVxlanTunnelMapKeysRecon.end();
+         it++)
+    {
+        std::string vlan, vlan_id, vni_id;
+        std::string vxlanTunnelMapName = *it;
+        std::vector<FieldValueTuple> temp; 
+        if (vxlanAppTunnelMapTable.get(vxlanTunnelMapName, temp))
+        {
+            for (auto fv: temp)
+            {
+                std::string field = fvField(fv);
+                std::string value = fvValue(fv);
+                SWSS_LOG_NOTICE("RESTORE Vxlan Tunnel MAP Table key: %s field: %s value: %s",
+                        vxlanTunnelMapName.c_str(), field.c_str(), value.c_str());
+                if (field == VLAN)
+                {
+                    vlan = value;
+                }
+                else if (field == VNI)
+                {
+                    vni_id = value;
+                }
+            }
+        }
+        else
+        {
+            SWSS_LOG_NOTICE("RESTORE VxLAN Tunnel Map Table Key(%s)", vxlanTunnelMapName.c_str());
+        }
+        const auto vlan_prefix = std::string("Vlan");
+        const auto prefix_len = vlan_prefix.length();
+        vlan_id = vlan.substr(prefix_len);
+
+        size_t found = vxlanTunnelMapName.find(delimiter);
+        const auto vxlanTunnelName = vxlanTunnelMapName.substr(0, found);
+
+        createVxlanNetdevice(vxlanTunnelName, vni_id, src_ip, dst_ip, vlan_id);
+        SWSS_LOG_NOTICE("RESTORE Created Kernel Net Device (%s-%s)", vxlanTunnelName.c_str(), vlan_id.c_str());
+    }
+
+    SWSS_LOG_NOTICE("RESTORE Delete Stale Kernel Net Devices");
+    clearAllVxlanDevices();
+    SWSS_LOG_NOTICE("RESTORE Recreate Kernel Cache");
+    getAllVxlanNetDevices();
+}
+
+void VxlanMgr::clearAllVxlanDevices()
+{
+    for (auto it = m_vxlanNetDevices.begin(); it != m_vxlanNetDevices.end();)
+    {
+        SWSS_LOG_NOTICE("Deleting Stale NetDevice vxlandevname %s\n", (it->first).c_str());
+        deleteVxlanNetdevice(it->first);
+        it = m_vxlanNetDevices.erase(it);
+    }
+}
+
+void VxlanMgr::waitTillReadyToReconcile()
+{
+    for(;;)
+    {
+        WarmStart::WarmStartState state;
+        WarmStart::getWarmStartState("vlanmgrd", state);
+
+        if ((WarmStart::REPLAYED == state) ||
```

Comment:
There is no `REPLAYED` state in the sonic-swss-common [PR339](https://github.com/Azure/sonic-swss-common/pull/339)

Reply:
Will add it in sonic-swss-common

Reply:
Separate PR on swss-common will be filed for new states added as part of warm reboot.